dotCMS/core
# fix(plugin): #29276 - Fix portlet serialization using immutables and jackson

### Proposed Changes
* JAXB Serialisation was causing issues due to the masking of fields from the base Portlet by DotPortlet.  When you reference the cast to Portlet you get the field on that class, and get different results when cast as DotPotlet.  Instead of extending Portlet in this way,  or tying the serialization to the underlying object we can make use of the immutables framework to create a serializable specific version of Portlet that can easily be mapped to and from the underlying Portlet.  This model can be used in many cases providing an abstraction layer that is more flexible to changes and easier to test. 

### Testing

* Portlets need to be able to load and should test the Plugins page loads and check the logs for error
* Add plugins including the cdn plugin and ensure it starts up correctly. 
* Add the CDN Plugin to the menus and browse to the plugin page.  It will show loading spinners and show an error in the log as it is not configured but the page headings should show and there should not be other errors.
* On stop of the plugin there may be an error.  This is expected when the jersey servlet container is restarted.  The plugin should be able to be started again after.
```[18/07/24 17:48:21:869 GMT]  WARN mapper.RuntimeExceptionMapper: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down
java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down```


related to #29276 

This PR fixes: #29276
